dockstore-integration-testing/src/test/java/io/dockstore/client/cli/CRUDClientIT.java

```diff
@@ -282,6 +282,25 @@ public void testRefreshingHostedTool() {
         });
     }
 
+    /**
+     * This tests that you cannot register a duplicate hosted tool
+     */
+    @Test
+    public void testHostedToolDuplicate() {
+        final ApiClient webClient = getWebClient(ADMIN_USERNAME);
+        HostedApi hostedApi = new HostedApi(webClient);
+        DockstoreTool hostedTool = hostedApi.createHostedTool("awesomeTool", "cwl", "quay.io", "coolNamespace");
+        boolean success = true;
+        try {
+            DockstoreTool hostedToolDuplicate = hostedApi.createHostedTool("awesomeTool", "cwl", "quay.io", "coolNamespace");
+        } catch (ApiException ex) {
+            success = false;
+        } finally {
```

Comment:
technically, putting the assert into a `finally` is redundant since the test will fail anyway if there is an uncaught exception